### PR TITLE
use home folder of current user

### LINKED ISSUE #15: "Hardcoded" PhoneMemory-folder makes /home/nemo practically inaccessible
@llewelld I was surprised to see, that I cannot simply select `/home/nemo` as my default folder in FileCase ≥ 0.2.1, but was relieved for a second when I found your commit 7369833 with its headline:
*Remove hardcoded references to /home/nemo
Moves the definition of the home folder into the settings file and defaults to /home/default.*

But then I discovered, that:
* One cannot change this setting at the GUI: This is an unsurmountable obstacle for many.
* [The new settings option is called "home"]( https://github.com/CepiPerez/filecase-sailfish/commit/7369833f7e30212f787d3173327c7280cfd0fe3d#diff-6b2f0a449fdefd8930e23ef0dcd752beec69242e1303d77653f047c5e0766385R32), which is in stark contrast to all other options being multi-word options in "CamelCase" and the term used at the GUI ("Phone memory").
* Any documentation or a simple description how to use this is lacking: I did that in the section "TL;DR" here.
* Side note: The [default value is `/home/defaultuser`](https://github.com/CepiPerez/filecase-sailfish/commit/7369833f7e30212f787d3173327c7280cfd0fe3d#diff-6b2f0a449fdefd8930e23ef0dcd752beec69242e1303d77653f047c5e0766385R32), not `/home/default` as denoted in the headline of commit 7369833.
* The unconditional introduction of a new default value (`/home/defaultuser`) for a new setting (`home`) breaks extant installations of FileCase, which were using `/home/nemo`, when updating them to any version ≥ 0.2.1.

IMO performing this change in such a manner is a bit "evil", because it will leave many helplessly stranded.  It took me more than 15 minutes of research and testing to comprehend how to utilise this correctly.

My criticism ultimately is, that this commit is only a half-step towards implementing a GUI option in FileCase's settings to make this option accessible for an average user.
My suggestion is to offer `/home/nemo` and `/home/defaultuser` as predefined values, but to allow for inputting any path.
<br />

Hence I suggested via PR #14 (merged) to alter settings option "home" to "PhoneMemoryFolder", because that is what it is called at the GUI, is more self-explanatory and is aligned with extant option names (e.g., in "CamelCase").

But this still leaves the ToDo open to implement a GUI option for setting this in FileCase's settings page.
<br />

#### TL;DR (documenting the extant workaround for others)
* One can set the default folder to be used as "Phone memory" folder by adding a line to `~/.config/cepiperez/fileboxplus.conf` in the `[General]` section since FileCase 0.2.1, e.g., `home=/home/nemo`
  For releases after PR #14 (i.e., FileCase > 0.4.0) this is, e.g., `PhoneMemoryFolder=/home/nemo`
* To achieve that with FileCase:
   1. One has to enable "Show root filesystem" and "Show hidden files" in FileCase's settings.
   2. Then ascend and descend into the real user's home directory (e.g., `/home/nemo`) and there into `.config/cepiperez`
   3. Tap on the file `fileboxplus.conf`, select "Rename" in the pulley menu and append the extension `.txt`
   4. Then tap on the file `fileboxplus.conf.txt`, select "Edit" in the pulley menu and add aforementioned line in the `[General]` section.
   5. Ultimately tap on the file `fileboxplus.conf.txt` again, select "Rename" in the pulley menu and remove the extension `.txt`, so that the file is called `fileboxplus.conf` again.

### PR DESCRIPTION
**Pull request by simonschmeisser**
closes #15 

This should return the current users home directory instead of hard coding it.

https://doc.qt.io/qt-5/qstandardpaths.html

[`QDir::homePath](https://doc.qt.io/qt-5/qdir.html#homePath) would be another alternative

(not actually tested)

### CODE REVIEW

**Comment by Olf0** [comment]
@simonschmeisser, thank you very much.

> This should return the current users home directory instead of hard coding it.

Originally I thought it would be nice, if users can set it to any path they like: which is already the case, but not yet via a UI dialogue.

But yesterday, when reading [the criticism at FSO](https://forum.sailfishos.org/t/sailfish-community-news-2nd-november-qmf/17169/2), I already considered to automatically set it to the user's homedir.  As you have come to this conclusion independently and provided a PR implementing this, this will be the way to go.  Probably this is the best solution, because one line of thinking (which should not be followed too strictly) is, that a configuration option is just a programmers easy way out, when unsure which default value to set.

**Review by Olf0 — APPROVED** [review]
LGTM

**Comment by simonschmeisser** [comment]
Maybe the best solution would be to use the home dir as default and let users change it if desired. I say maybe because I haven't actually used this app just thought this should be easy to fix and went for it
